lib/kubeclient.rb

```diff
@@ -202,5 +202,11 @@ def all_entities
         result_hash[key_name] = send(method_name)
       end
     end
+
+    def api_versions_on_server
+      versions_endpoint = URI.parse(@api_endpoint.to_s)
+      versions_endpoint.path.slice! @api_version
+      RestClient::Resource.new(versions_endpoint.to_s).get
```

Comment:
There are too many conversions in these three lines  (and a string manipulation), I feel like we should keep the api_endpoint (as uri) and api_version separate and join them only before a rest query. Best case here would be to do just:

    RestClient::Resource.new(@api_endpoint).get